contracts/DexX.sol

```diff
@@ -0,0 +1,89 @@
+// contracts/DexX.sol
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import "@redstone-finance/evm-connector/contracts/data-services/MainDemoConsumerBase.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract DexX is MainDemoConsumerBase, Ownable {
+
+    bytes32 public constant AVAX_SYMBOL = bytes32("AVAX");
+
+    event NewOracleDataRequest(
+        uint256 indexed avaxToSwap,
+        address indexed requestedBy,
+        uint256 indexed requestedAtBlock
+    );
+
+    ERC20 public usd;
+    mapping (bytes32 => bool) requestedSwaps;
+
+    constructor(ERC20 usdToken) Ownable() {
+        usd = usdToken;        
+    }
+
+    // This function is called by a user to request a swap
+    // It doesn't require attaching a redstone payload
+    function changeAvaxToUsd() external payable {
+        bytes32 requestHash = calculateHashForSwapRequest(msg.value, msg.sender, block.number);
+        requestedSwaps[requestHash] = true;
```

Comment:
Let's throw if requestedSwaps[requestHash] == true
(Someone created two requests in a single block, i.e. double click)

Reply:
Ok